scripts/checkpatch.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+CHECKPATCH=checkpatch.pl
+
+function _checkpatch() {
+		$CHECKPATCH --quiet --ignore FILE_PATH_CHANGES \
+				--ignore GERRIT_CHANGE_ID --no-tree -
+}
+
+function checkpatch() {
+		echo "Checking commit:  "
+		git show --oneline --no-patch $1
+		git format-patch -1 $1 --stdout -- . $_CP_EXCL | _checkpatch
+}
+
+function checkstaging() {
+		echo "Checking staging area:  "
+		git diff --cached | _checkpatch
+}
+
+function checkdiff() {
+		echo "Checking squashed commits (diff $1...$2)"
+		git diff $1...$2 -- . $_CP_EXCL | _checkpatch
+}
+
+hash $CHECKPATCH 2>/dev/null ||
+		{ echo >&2 "Could not find checkpatch.pl, aborting"; exit 1; }
+
+case "$1" in
+	--cached)
+			checkstaging
+			;;
+	--commit)
+			checkpatch $2
+			;;
+	--diff)
+			checkdiff $2 $3
+			;;
+	*)
+			echo "Usage: $0 { --cached| --commit <commit_hash>|"\
+			"--diff <commit1_hash> <commit2_hash>}"
```

Comment:
- Syntax: alternatives should be specified with square brackets and you're missing spaces before the `|`.
- Use `<commit>` instead of `<commit_hash>` because a branch name or tag will do, for instance.
- With the proposed help string, it is not obvious that you can't call the script without an argument. Also it would be good to make `--commit` optional, just pass the commit identifier. And even better to accept several commits (otherwise drop the `...`). So here is what I suggest:
```
Usage: $0 <commit>...
       $0 --diff <commit1> <commit2>
       $0 --cached
```